### ISSUE TITLE
关于DIN的疑问

### ISSUE THREAD

**Issue by yanduoduan** [Prayforhanluo/CTR_Algorithm#4]
大佬您好，最近看您的DIN代码，有一些地方不太明白，希望得到您的解答！
1、mask = (behaviors_x > 0).float().unsqueeze(-1) 这里的msak的具体作用是啥啊？为什么需要这个mask呢？
这里的 注意力输入部分，原始的好像没有queries - user_behavior吧？为啥有这一项呢
2、attn_input = torch.cat([queries, user_behavior, 
                                queries - user_behavior, 
                                queries * user_behavior], dim = -1)
3、 output = user_behavior.mul(attns.mul(mask)) # batch * seq_len * embed_dim
这个里面为啥还有mask呢？
